satellite/gracefulexit/endpoint.go

```diff
@@ -445,34 +363,16 @@ func (endpoint *Endpoint) doProcess(stream processStream) (err error) {
 			if err != nil {
 				if ErrInvalidArgument.Has(err) {
 					// immediately fail and complete graceful exit for nodes that fail satellite validation
+					mon.Meter("graceful_exit_fail_validation").Mark(1)
 					exitStatusRequest := &overlay.ExitStatusRequest{
 						NodeID:         nodeID,
 						ExitFinishedAt: time.Now().UTC(),
 						ExitSuccess:    false,
 					}
-
-					finishedMsg, err := endpoint.getFinishedMessage(ctx, endpoint.signer, nodeID, exitStatusRequest.ExitFinishedAt, exitStatusRequest.ExitSuccess, pb.ExitFailed_VERIFICATION_FAILED)
-					if err != nil {
-						return rpcstatus.Error(rpcstatus.Internal, err.Error())
-					}
-
-					mon.Meter("graceful_exit_fail_validation").Mark(1)
-					_, err = endpoint.overlaydb.UpdateExitStatus(ctx, exitStatusRequest)
+					err := endpoint.handleFinished(ctx, stream, exitStatusRequest, pb.ExitFailed_VERIFICATION_FAILED)
 					if err != nil {
-						return rpcstatus.Error(rpcstatus.Internal, err.Error())
+						return err
```

Comment:
the errors inside `handleFinished` are all rpcstatus wrapped with the internal error, but I think it would be more consistent with the rest of this function to wrap this err here with rpc, and just return regular err in `handleFinished`